pkg/config/store_test.go

```diff
@@ -0,0 +1,383 @@
+package config
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/stacklok/toolhive/pkg/env"
+	"github.com/stacklok/toolhive/pkg/state"
+)
+
+func TestKubernetesStore_Load(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	t.Run("load existing config", func(t *testing.T) {
+		t.Parallel()
+
+		client := fake.NewSimpleClientset()
+		store := &KubernetesStore{
+			client:     client,
+			namespace:  "toolhive-system", // Use toolhive-system for testing
+			serverName: "test-server",     // Use a test server name
+		}
+
+		// Create a ConfigMap with config data
+		configData := `secrets:
+  provider_type: "encrypted"
+  setup_completed: true
+clients:
+  registered_clients: []
+  auto_discovery: false
+registry_url: "https://registry.example.com"
+local_registry_path: ""
+allow_private_registry_ip: false
+default_group_migration: false`
+
+		configMap := &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "toolhive-config-test-server", // Server-specific ConfigMap name
+				Namespace: "toolhive-system",
+				Labels: map[string]string{
+					"app.kubernetes.io/name":      "toolhive",
+					"app.kubernetes.io/component": "config",
+				},
+			},
+			Data: map[string]string{
+				ConfigMapKey: configData,
+			},
+		}
+
+		_, err := client.CoreV1().ConfigMaps("toolhive-system").Create(ctx, configMap, metav1.CreateOptions{})
+		require.NoError(t, err)
+
+		// Load config
+		config, err := store.Load(ctx)
+		require.NoError(t, err)
+		require.NotNil(t, config)
+
+		assert.Equal(t, "encrypted", config.Secrets.ProviderType)
```

Comment:
I'm sure this is just a dummy test value, but encrypted secrets isn't a secrets provider inside of Kubernets